src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

```diff
@@ -3540,6 +3540,34 @@
     public static final int COMMENT_CONTENT =
             GeneratedJavaTokenTypes.COMMENT_CONTENT;
 
+    /**
+     * A pattern variable definition; when conditionally matched,
+     * this variable is assigned with the defined type.
+     *
+     * <p>For example:</p>
+     * <pre>
+     * if (obj instanceof String str) { }
+     * </pre>
+     * <p>parses as:</p>
+     * <pre>
+     * LITERAL_IF (if)
+     *  |--LPAREN (()
+     *  |--EXPR
+     *  |   `--LITERAL_INSTANCEOF (instanceof)
+     *  |       `--IDENT (obj)
+     *  |           `--PATTERN_VARIABLE_DEF
```

Comment:
I do not understand why PATTERN appear under IDENT, it should be sibling.

```
$ cat Test.java 
public class Test {
 void foo() {
  String s = "";
  boolean b = s instanceof String;
 }
}

$ java $RUN_LOCALE -jar /var/tmp/checkstyle-8.33-all.jar -t Test.java | grep -C 3 instanceof | sed "s/    |       |   //"
|--IDENT -> b [4:10]
`--ASSIGN -> = [4:12]
    `--EXPR -> EXPR [4:16]
        `--LITERAL_INSTANCEOF -> instanceof [4:16]
            |--IDENT -> s [4:14]
            `--TYPE -> TYPE [4:27]
                `--IDENT -> String [4:27]
```

even in txt inputs you have them as sibling.